source/controller/project_controller.py

```diff
@@ -253,6 +258,69 @@ def update_graph_to_table(self, edited_graph):
     def to_line_graph(self):
         self.visualize_graph_dock.plot_graph(nx.line_graph(self.visualize_graph_dock.current_graph))
 
+    def on_save_graph(self):
+        line_text = str(self.project_tool_bar.combo_graphs.currentText()[10:]) + "\n"
+        current_index = self.project_tool_bar.combo_graphs.currentIndex()
+
+        file_path = str(project_information_store.file_path)
+        file_path = file_path[2:-3]
+        file_type = file_path[len(file_path) - 1]
+        new_g6 = ""
+        replaced_line = ""
+
+        if self.edited_graph == None and self.edited_graph != line_text:
+            new_g6 = line_text[:-1]
+        else:
+            new_g6 = nx.to_graph6_bytes(self.edited_graph)[10:-1].decode("utf-8")
+
+        if file_type == "6" or file_type == "t":
+            file = open(file_path, "r")
+            for line in file:
```

Comment:
You can replace a specific line using readlines(), like [here](https://www.geeksforgeeks.org/python-program-to-replace-specific-line-in-file/).